src/gallium/drivers/lima/lima_resource.c

```diff
@@ -207,6 +222,18 @@ lima_resource_from_handle(struct pipe_screen *pscreen,
    res->stride = handle->stride;
    res->buffer->screen = screen;
 
+   if (screen->ro) {
```

Comment:
It should be not necessary to export/import every imported resources to sun4i. As my understanding, this chunk of code is for the following lima_resource_get_handle be able to call renderonly_get_handle, but we can just use the lima_bo_export, so this chunk is not needed or only needed when templat->bind & PIPE_BIND_SCANOUT like lima_resource_create. Can you confirm this?

Reply:
This was indeed not necessary for the renderonly support in lima. Renderonly will only allocate buffers in lima and export them, so this does not get called.